youtube_dl/extractor/vier.py

```diff
@@ -45,6 +49,24 @@ class VierIE(InfoExtractor):
         'only_matching': True,
     }]
 
+    def _real_initialize(self):
+        self._login()
+
+    def _login(self):
+        username, password = self._get_login_info()
+        if username is None:
+            return
+
+        self._request_webpage(
+            'http://www.vier.be/user/login',
```

Comment:
@dstftw: Because `_login` doesn't have access to the URL, we don't know here whether we're dealing with vier.be or vijf.be, so I *temporarily* hardcoded vier.be here.

How do other extractors deal with this? I couldn't find anything by grepping for `<site>` + `login`.

My current plan: add `site` as parameter to the `_login` method (rename `_login` to something else?) and call it in `_real_extract` after we have extracted the `site`. Furthermore, set and check `self._logged_in` so authentication happens at most once.